[WIP] Tests: Mark skipped tests as pending (with reason)

janpio opened the pull request:
Skip examples that don't work on non-macOS platforms instead of filtering them (which marks and reports them as pending). also offer reasons for skipping.

Gives output:

> [17:45:21]: ▸ 4704 examples, 0 failures, 175 pending

(Implementation details via discussion at http://groups.google.com/forum/#!topic/rspec/5qeKQr_7G7k)

(has to be rebased after #11242 is merged or updated)

janpio committed:
Switch from macOS to Ubuntu-Ruby-Docker image

janpio committed:
new cache key as old cache can't be restored

janpio committed:
no brew on ubuntu, use apt-get

janpio committed:
change default/ENV encoding

janpio committed:
set FASTLANE_ITUNES_TRANSPORTER_PATH to existing path (of non existing iTMSTransporter installation)

janpio committed:
download, build and install xar

janpio committed:
combine macos and ubuntu config.yml
name jobs
add workflows to parallelize both

janpio committed:
skip examples that don't work on non-macOS platforms instead of filtering them (which marks and reports them as pending)
also offer reasons for skipping
(via https://groups.google.com/forum/#!topic/rspec/5qeKQr_7G7k)

fastlane-bot-helper commented:
<!--
  0 Errors
  1 Warning: Pull Request is Work in Progre...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

janpio committed:
also skip before(:all) by overriding the hooks (hook is skipped if parent has metadata that indicates a skip)

janpio committed:
better text and comments

janpio committed:
better cache keys
moved installation of additional dependency outside of cache "bracket" for more logic

janpio committed:
login shell not necessary on ubuntu

janpio committed:
ruby version not necessary on ubuntu

janpio committed:
updating gems not necessary on ubuntu

janpio committed:
fix test reports stuff again (as only half the file got it at rebase)

janpio committed:
rename "test-results" to "test-reports" (which corresponds to the suggested Circle CI env var) to avoid future confusion

janpio commented:
replaced by #11320
